### ISSUE TITLE
Filter: opened today?

### ISSUE THREAD

**Issue by florianwittmann** [VeggieSailor/VeggieSailor#6]
Similar to setting a filter for minimum veg-type (vegan / vegetarian and so on) I want to filter for restaurant that are open today.
It should show only restaurants that are open now or in the future today, but not today closed restaurants or already closed ones.

**Comment by bluszcz** [comment]
Good idea :+1:

**Comment by florianwittmann** [comment]
I'm working on it.

**Comment by bluszcz** [comment]
BTW please check my latest colors update: https://github.com/bluszcz/VeggieSailor/commit/f962988480eca179449c8a0b94b53c95b1d03ec1

**Comment by florianwittmann** [comment]
Yes noticed that one ;-) Currently merging. Opening hours is little bit harder as the object from api is not very cool to parse. Will do that later. Now preparing this small refactor + city page improvements pullrequest for now.

**Comment by bluszcz** [comment]
I think it would be good to have date parsing made in model - python part done

**Comment by florianwittmann** [comment]
Makes sense :+1: 
Would be good to have a easy to parse date model in qml/js.
Something like: Array of 7 days (Starting day monday?) with array of timeareas in 24h time.
[ [{from:1000, to:1400}, {from:1600, to:2100}],  \that's monday
 [{from:1000, to:1400}, {from:1600, to:2100}], \thuesday and so on
 [] // Wednesday closed --> empty array of timeareas
..
]

**Comment by bluszcz** [comment]
Ok, I ill look into it.

**Comment by bluszcz** [comment]
`hours_parsed` is available now.

**Comment by florianwittmann** [comment]
Working on the frontend filter part.

**Comment by florianwittmann** [comment]
hours_parsed for a entry just contains: {"objectName":"","count":7,"dynamicRoles":false}
instead of the array.

**Comment by florianwittmann** [comment]
hours_parsed seems to be an QQmlListModel and not a simple plain javascript array. JSON.stringify resolves it just to: {"objectName":"","count":7,"dynamicRoles":false}

**Comment by bluszcz** [comment]
Hm, checking.

**Comment by bluszcz** [comment]
Not sure if pyotherside makes good translation nested lists from python to javascript...

On python side it looks as correct result, not sure what is wrong:

```
$ python3 listmodel.py 
[{'days': 'Daily', 'hours': ['8pm - 12:30pm']}]

gives 

 [[{'to': '2430', 'from': '2000'}], [{'to': '2430', 'from': '2000'}], [{'to': '2430', 'from': '2000'}], [{'to': '2430', 'from': '2000'}], [{'to': '2430', 'from': '2000'}], [{'to': '2430', 'from': '2000'}], [{'to': '2430', 'from': '2000'}]]
```

**Comment by bluszcz** [comment]
Actually I have just debug it and JSON.stringify  says:

```
hours_parsed":[[{"from":"1100","to":"2400"}],[{"from":"1100","to":"2400"}],[{"from":"1100","to":"2400"}],[{"from":"1100","to":"2400"}],[{"from":"1100","to":"2400"}],
[{"from":"1100","to":"2400"}],[{"from":"1100","to":"2400"}]]
```

I put in Entries.qml:

```
    function showEntries(entries) {
        loadingData = false;
        listModel.clear();
        for (var i=0; i<entries.length; i++) {
            if(entries[i].veg_level >= minimumVegLevel) {
                listModel.append(entries[i]);
                console.log(JSON.stringify((entries[i])));
            }

        }
```

So I think it is working and ready to frontend implementation.

**Comment by florianwittmann** [comment]
Cool, will look into this these days! :+1:
